### ISSUE TITLE
"make generate" consultation to add grafana dashboards

### ISSUE THREAD

**Issue by eedugon** [coreos/prometheus-operator#532]
Hello,

Sorry if this question looks too simple or basic, but I have tried to find within the project documentation the information to be able to include some dashboards in grafana / grafana-watcher, and what I found is that I need to put the dashboards into the assets/grafana directory and then run "make generate", so the new configmap will be generated. That looks great, but...

Would you please let me know what do I need to setup in a generic linux system to make the "make generate" command to work? I'm pretty sure I will need some kind of "go compiler" and "make", but: any specific version? some environment variables set up? from what directory exactly should I run the make command?

If you have all that documented, then just share a link with me, and sorry for bothering with this :)

If compiling stuff is going to be part of the user's normal activities I think you should have that documented somewhere. And being able to deploy grafana dashboards I think it's important, otherwise after every restart of grafana we have to import them manually...

Thanks a lot and regards!
Edu

**Comment by brancz** [comment]
I think we have not documented which directory to run `make generate` in. As the dependencies are actually quite small: bash, cat, echo, base64. Are you running `make generate` in the root directory of the repository or in `contrib/kube-prometheus/`? The `Makefile` to generate the manifests is in `contrib/kube-prometheus/Makefile`, does that make sense? Do you think we need to improve the docs?

**Comment by eedugon** [comment]
oh, my bad! I didn't see the Makefile because I wasn't updated to the latest master commit, and in my repository that file wasn't there, I'm sorry =) so, i didn't find the "generate" definition anywhere.

Yes, the doc then is accurate, maybe just adding a comment about where the make generate should be run. I will add that comment and raise a small PR for your validation.

Now I have also realized that the scripts to generate the configMap are there and are very simple too.
I have created something that does almost the same, but giving the option of applying the new configmap into kubernetes directly and restart grafana. I can share it if you want.

I'm closing this issue, as there was no issue actually :)
